cmake/FindCMSIS.cmake

```diff
@@ -77,9 +77,13 @@ foreach(COMP ${CMSIS_FIND_COMPONENTS})
 
     if((NOT STM32_CMSIS_${FAMILY}_PATH) AND (NOT STM32_CUBE_${FAMILY}_PATH))
         set(STM32_CUBE_${FAMILY}_PATH /opt/STM32Cube${FAMILY} CACHE PATH "Path to STM32Cube${FAMILY}")
-        message(STATUS "Neither STM32_CUBE_${FAMILY}_PATH nor STM32_CMSIS_${FAMILY}_PATH specified using default  STM32_CUBE_${FAMILY}_PATH: ${STM32_CUBE_${FAMILY}_PATH}")
+        message(STATUS "Neither STM32_CUBE_${FAMILY}_PATH nor STM32_CMSIS_${FAMILY}_PATH specified using default STM32_CUBE_${FAMILY}_PATH: ${STM32_CUBE_${FAMILY}_PATH}")
     endif()
-        
+
+    if ((${CMSIS_FIND_REQUIRED_${COMP}}) AND (NOT EXISTS ${STM32_CMSIS_${FAMILY}_PATH}) AND (NOT EXISTS ${STM32_CUBE_${FAMILY}_PATH}))
+        message(WARNING "For REQUIRED COMPONENT: ${COMP}, Neither STM32_CUBE_${FAMILY}_PATH: '${STM32_CUBE_${FAMILY}_PATH}' nor STM32_CMSIS_${FAMILY}_PATH: '${STM32_CMSIS_${FAMILY}_PATH}' Exists")
```

Comment:
From the cmake documentation, QUIET only applies to OPTIONAL packages. That block only prints an error for REQUIRED_COMPONENTS so QUIET shouldn't apply.

But I'll make sure other message statements only print if REQUIRED or NOT QUIET.

Reply:
On second thought, considering the number of messages printed, I believe this should be broken out into a separate feature so that it gets handled consistently in the package.

Reply:
I understand this as 
> The QUIET option disables informational messages, including _**but not limited to**_ those indicating that the package cannot be found if it is not REQUIRED.

But it's open to interpretation

More info in [cmake-developper page](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html).
From my understanding `QUIET` avoids all print excluding final error in case of `REQUIRED` is used. 
Well I just wanted to heads up but it's not so important from my point of view as long as we keep it homogeneous.

Reply:
> On second thought, considering the number of messages printed, I believe this should be broken out into a separate feature so that it gets handled consistently in the package.

No problem. I agree managing only initial point here is better practice. 
I opened #247 but this is absolutely not forcing you to do the work, just keeping track of it. Of course we would be happy if you contribute.